namerd/iface/destination/src/main/scala/io/buoyant/namerd/iface/destination/DestinationService.scala

```diff
@@ -0,0 +1,190 @@
+package io.buoyant.namerd.iface.destination
+
+import com.twitter.finagle.naming.NameInterpreter
+import com.twitter.finagle.{Address, _}
+import com.twitter.io.Buf
+import com.twitter.logging.Logger
+import com.twitter.util.{Event, _}
+import io.buoyant.grpc.runtime.VarEventStream.{Ev, Val}
+import io.buoyant.grpc.runtime.{GrpcStatus, Stream, VarEventStream}
+import io.linkerd.proxy.destination.Destination
+import io.linkerd.proxy.destination.Update.OneofUpdate
+import io.linkerd.proxy.destination.Update.OneofUpdate.{NoEndpoints => EndpointsNone, _}
+import io.linkerd.proxy.destination._
+import io.linkerd.proxy.net.IPAddress.OneofIp
+import io.linkerd.proxy.net.{IPAddress, TcpAddress}
+import java.net.{InetAddress, InetSocketAddress}
+import java.util.concurrent.atomic.AtomicBoolean
+
+class DestinationService(
+  pathPfx: String,
+  interpreter: NameInterpreter
+) extends Destination {
+  import DestinationService._
+
+  //Helper trait to capture Failed Addresses
```

Comment:
I think this can be cleaned up if you define `type TrySet[T] = Try[Set[T]]` and turn your Var[Addr] into an Event[TrySet[Address]].  This avoids the need for the FAddress class.

You would then define a Diffable[TrySet] which pattern matches on the left and right TrySets.  If both are return, it returns a SetDiff as usual.  If either of them are Throw then it returns a new class called Failed[T] or something like that which extends Diff[TrySet, T].